File: compiler/testData/asJava/lightClasses/lightClassByPsi/k2/jvmExposeBoxed/noArg4.java
@kotlin.OptIn(markerClass = {kotlin.ExperimentalStdlibApi.class})
@kotlin.jvm.JvmInline()
public final class IntWrapper /* IntWrapper*/ {
  private final int i;

  @kotlin.jvm.JvmExposeBoxed()
  public  IntWrapper();//  .ctor()

  @kotlin.jvm.JvmExposeBoxed()
  public  IntWrapper(int);//  .ctor(int)

  @org.jetbrains.annotations.NotNull()
  public @org.jetbrains.annotations.NotNull() java.lang.String toString();//  toString()

  public boolean equals(@org.jetbrains.annotations.Nullable() @org.jetbrains.annotations.Nullable() java.lang.Object);//  equals(@org.jetbrains.annotations.Nullable() java.lang.Object)

  public final int getI();//  getI()

  public int hashCode();//  hashCode()
}

public final class RegularClassWithValueConstructor /* RegularClassWithValueConstructor*/ {
  private final int property;

  private  RegularClassWithValueConstructor(int);//  .ctor(int)
}

@kotlin.OptIn(markerClass = {kotlin.ExperimentalStdlibApi.class})
public final class RegularClassWithValueConstructorAndAnnotation /* RegularClassWithValueConstructorAndAnnotation*/ {
  private final int property;

  @kotlin.jvm.JvmExposeBoxed()
  public  RegularClassWithValueConstructorAndAnnotation();//  .ctor()

  @kotlin.jvm.JvmExposeBoxed()
  public  RegularClassWithValueConstructorAndAnnotation(@org.jetbrains.annotations.NotNull() IntWrapper);//  .ctor(@org.jetbrains.annotations.NotNull() IntWrapper)

  private  RegularClassWithValueConstructorAndAnnotation(int);//  .ctor(int)
}

File: compiler/testData/asJava/lightClasses/lightClassByPsi/k2/jvmExposeBoxed/noArg4.kt
// WITH_STDLIB
// LIBRARY_PLATFORMS: JVM

@OptIn(ExperimentalStdlibApi::class)
@JvmInline
value class IntWrapper @JvmExposeBoxed constructor(val i: Int = 0)

class RegularClassWithValueConstructor(val property: IntWrapper = IntWrapper(1))

@OptIn(ExperimentalStdlibApi::class)
class RegularClassWithValueConstructorAndAnnotation @JvmExposeBoxed constructor(val property: IntWrapper = IntWrapper(2))

// LIGHT_ELEMENTS_NO_DECLARATION: IntWrapper.class[constructor-impl;equals-impl;equals-impl0;hashCode-impl;toString-impl], RegularClassWithValueConstructor.class[getProperty-7j0DjTs], RegularClassWithValueConstructorAndAnnotation.class[getProperty-7j0DjTs]